### ISSUE TITLE
Consider battery degredation in Prop_DeviceBatteryPercentage_Float

### ISSUE THREAD

**Issue by Curtis-VL** [ValveSoftware/openvr#1499]
Currently when getting the Prop_DeviceBatteryPercentage_Float value this is before battery degridation has been taken into account, devices can shutdown at as high as 15% over time.

Including battery degredation would be great assuming it's possible to determine that in some way!

**Comment by okawo80085** [comment]
this prop is handled by drivers, it should be the vendor's responsibility to accommodate for that, direct input from the vendor will be needed to properly correct for it anyway, so might as well make the vendors do it(if they are not doing it already) +no changes to a working api

**Comment by Curtis-VL** [comment]
> this prop is handled by drivers, it should be the vendor's responsibility to accommodate for that, direct input from the vendor will be needed to properly correct for it anyway, so might as well make the vendors do it(if they are not doing it already) +no changes to a working api

This is a very good point! But it also reminds me that this may never happen. I don't see manufacturers changing this on their end.
HTC never did it, Valve doesn't do it, and of the other companies that actually provide battery life (Everyone but Oculus) don't do it either.

**Comment by okawo80085** [comment]
yeah, as a driver maintainer myself battery charge degradation, and battery charge monitoring in general, is one of my last concerns

**Comment by JoeLudwig** [comment]
This is certainly a feature for drivers if they have enough information to provide it. SteamVR doesn't know enough about any devices other than Index controllers to provide these numbers.

Have you see an issue with this for Index controllers?

**Comment by Curtis-VL** [comment]
> This is certainly a feature for drivers if they have enough information to provide it. SteamVR doesn't know enough about any devices other than Index controllers to provide these numbers.
> 
> Have you see an issue with this for Index controllers?

I haven't check Index yet, the issue was mainly with devices like the Vive Trackers which have very short battery life. I'd say most users don't hit 0% on their Index controllers.

Vive Trackers for example, depending on battery health, die anywhere between 5% and 12% of what I understand from most users I've spoke to. This is moreso the issue.
